ktesting/testinglogger_test.go

```diff
@@ -145,3 +149,37 @@ func (l *logToBuf) Log(args ...interface{}) {
 	}
 	l.Write([]byte("\n"))
 }
+
+func TestStop(t *testing.T) {
+	// This test is set up so that a subtest spawns a goroutine and that
+	// goroutine logs through ktesting *after* the subtest has
+	// completed. This is not supported by testing.T.Log and normally
+	// leads to:
+	//   panic: Log in goroutine after TestGoroutines/Sub has completed: INFO hello world
+	//
+	// It works with ktesting if (and only if) logging gets stopped
+	// before returning from the test.
+	var wg1, wg2 sync.WaitGroup
+	wg1.Add(1)
+	wg2.Add(1)
+	t.Run("Sub", func(t *testing.T) {
+		logger, _ := ktesting.NewTestContext(t)
+		go func() {
+			defer wg2.Done()
+
+			// Wait for test to have returned.
+			wg1.Wait()
+
+			// This output must go to klog because the test has
+			// completed.
+			logger.Info("simple info message")
+			logger.Error(nil, "error message")
+			logger.WithName("me").WithValues("completed", true).Info("complex info message", "anotherValue", 1)
```

Comment:
I don't have an automated test for this. Not sure whether one is needed? Probably... anyway, currently it works:
```
go test -v -run=TestStop .
=== RUN   TestStop
=== RUN   TestStop/Sub
I0630 09:06:10.218591  113165 testinglogger_test.go:175] "TestStop/Sub leaked goroutine: simple info message"
E0630 09:06:10.218699  113165 testinglogger_test.go:176] "TestStop/Sub leaked goroutine: error message"
I0630 09:06:10.218722  113165 testinglogger_test.go:177] "TestStop/Sub leaked goroutine/me: complex info message" completed=true anotherValue=1
--- PASS: TestStop (0.00s)
    --- PASS: TestStop/Sub (0.00s)
PASS
ok  	k8s.io/klog/v2/ktesting	(cached)
```

Reply:
Done.